File: Java/Easy/1512.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int answer = 0;
        int temp;

        for (int i : nums) {
            temp = map.getOrDefault(i, 0);
            answer += temp;
            temp++;
            map.put(i, temp);
        }

        return answer;
    }
}

File: C/Easy/1512.c
int numIdenticalPairs(int *nums, int numsSize)
{
    // create hashtable
    int *hashtable = malloc(101 * (sizeof(int)));
    for (int i = 0; i < 101; i++)
    {
        hashtable[i] = 0;
    }

    int temp;

    int answer = 0;

    // iterate through the nums array
    for (int j = 0; j < numsSize; j++)
    {
        temp = nums[j];
        if (hashtable[temp])
        { // exist before
            answer += hashtable[temp];
            hashtable[temp]++;
        }
        else
        {
            // doesnt exist yet
            hashtable[temp] = 1;
        }
    }

    return answer;
}